src/metadataStorage.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Persist SDK event metadata
+ * Uses cookie if available, otherwise fallback to localstorage.
+ */
+
+import Base64 from './base64';
+import Cookie from './cookie';
+import localStorage from './localstorage'; // jshint ignore:line
+
+class MetadataStorage {
+  constructor({storageKey, disableCookies}) {
+    this.storageKey = storageKey;
+    this.disableCookieStorage = !Cookie.areCookiesEnabled() || disableCookies;
+  }
+
+  save({ deviceId, userId, optOut, sessionId, lastEventTime, eventId, identifyId, sequenceNumber }) {
+    // do not change the order of these items
+    const value = `${deviceId}.${Base64.encode(userId || '')}.${optOut ? '1' : ''}.${sessionId}.${lastEventTime}.${eventId}.${identifyId}.${sequenceNumber}`;
```

Comment:
the device id can also be overriden by the customer and can also be set to someone's email, should it be Base64 encoded too?

Reply:
I'll add an override if someone uses setDeviceId